### PR TITLE
Domains: fix warnings

### PR DESCRIPTION
**Pull request by dzver**
Fix unmatched propType 

To test:
1. Go to domains/add/site
2. Search for something

There shoudln't be any warnings

### CODE REVIEW

**Comment by blowery** [comment]
:eyeglasses: lookin

**Comment by blowery** [comment]
few warnings from eslint in the files changed by the branch:

```
  268:5   warning  Expected indentation of 3 tab characters but found 4  indent
  273:5   warning  Expected indentation of 3 tab characters but found 4  indent
  277:5   warning  Expected indentation of 3 tab characters but found 4  indent
  281:4   warning  Expected indentation of 2 tab characters but found 3  indent
  304:7   warning  JSX props should not use .bind()                      react/jsx-no-bind
  332:44  warning  '?' should be placed at the beginning of the line     operator-linebreak
  335:8   warning  ':' should be placed at the beginning of the line     operator-linebreak
  372:48  warning  '?' should be placed at the beginning of the line     operator-linebreak
  373:70  warning  ':' should be placed at the beginning of the line     operator-linebreak
```

Any chance we can get those fixed too?

**Comment by dzver** [comment]
didn't want to mix indentation fixes with code

**Comment by blowery** [comment]
I'd rather just see it fixed. We have plenty of ways to isolate whitespace changes when looking at a diff.

**Comment by blowery** [comment]
The search results when I switch sites seem to be for the _previously picked_ site, not the new one.

File: client/components/domains/register-domain-step/index.jsx [hunk 1]
```diff
@@ -82,8 +82,8 @@ var RegisterDomainStep = React.createClass( {
 		}
 	},
 
-	componentDidUpdate: function( prevProps ) {
-		if ( this.props.selectedSite && this.props.selectedSite.domain !== prevProps.selectedSite.domain ) {
+	componentWillReceiveProps: function( nextProps ) {
+		if ( this.props.selectedSite && this.props.selectedSite.domain !== nextProps.selectedSite.domain ) {
 			this.setState( this.getInitialState() );
```
Comment (blowery) [inline client/components/domains/register-domain-step/index.jsx]: probably need to pass the nextProps here to get the correct value

**Comment by blowery** [comment]
Ah yeah, I run into this problem a lot myself. When you're using componentWillReceiveProps, this.props have not yet updated to the new value, so you have to pass them to functions that expect to work with the props. I often have functions that looks like:

```
doAThing( props = this.props ) {
 // do the thing
}

componentWillReceiveProps( nextProps ) {
  this.doAThing( nextProps);
}

somethingElse() {
  this.doAThing();
}
```

File: client/components/domains/register-domain-step/index.jsx [hunk 2]
```diff
@@ -82,8 +82,8 @@ var RegisterDomainStep = React.createClass( {
...
 			this.focusSearchCard();
 			this.fetchDefaultSuggestions();
```
Comment (blowery) [inline client/components/domains/register-domain-step/index.jsx]: actually, here. This is what's using the old props and fetching suggestions for it.

File: client/components/domains/register-domain-step/index.jsx (hunk 1 above)
Comment (blowery) [inline client/components/domains/register-domain-step/index.jsx]: nope, i was wrong, i think

File: client/components/domains/register-domain-step/index.jsx (hunk 2 above)
Comment (dzver) [inline client/components/domains/register-domain-step/index.jsx]: I'll figure it out on Monday :)

Comment (blowery) [inline client/components/domains/register-domain-step/index.jsx]: works :)

**Comment by dzver** [comment]
Switching componentDidUpdate with componentWillReceiveProps seems to be a major fix. I'd like to commit this little change first, so that it doesn't wait.

**Comment by sirbrillig** [comment]
LGTM!

**Comment by sirbrillig** [comment]
Fixes #2772

**Pull request merged**